### DISCUSSION TITLE
numpexpr issue

### DISCUSSION THREAD

**Discussion by spandy007** [NatLabRockies/floris#511] [Q&A]
Hi, I have just cloned the Floris and was running example 1 but it looks like there is a package numpexpr that is causing an issue. Even after installing the package, I get the error. I also tried pip install numexpr but the issue is not solved. Could you let me know if there is some other way to fix this issue?

ModuleNotFoundError: No module named 'numexpr'

**Comment by paulf81** [comment]
My suspicion is that there is an environment issue where floris and numexpr aren’t in the same environment (or an environment which previously FLORIS V2 was installed).  Installing FLORIS itself should cause the installation of numexpr, could you try setting up FLORIS in a clean environment (I use conda myself for this) via:

Developers and anyone who intends to inspect the source code can install FLORIS by downloading the git repository from GitHub with git and use pip to locally install it. It is highly recommended to use a Python virtual environment manager such as conda (https://docs.conda.io/en/latest/miniconda.html) in order to maintain a clean and sandboxed environment. The following commands in a terminal or shell will download and install FLORIS.

```
# Download the source code from the `main` branch
git clone -b main https://github.com/NREL/floris.git

# If using conda, be sure to activate your environment prior to installing
# conda activate <env name>

# If using pyOptSpare, install it first
conda install -c conda-forge pyoptsparse

# Install FLORIS
pip install -e floris
```

This should resolve it,

Paul

**Comment by rafmudaf** [comment]
@spandy007 in the same line as @paulf81's comment, could you verify that your python environment indeed does have numexpr installed? There are two things to check. First, start a REPL (run-execute-print-loop) in your terminal by simply typing `python`. Then, try to import numexpr with `import numexpr`. This should fail with the same error. If it doesn't, then there's some mixup with python interpreters and/or Python environments (like conda, pyenv or venv). If that does fail, the check that you've install numexpr with pip: `pip freeze | grep numexpr`, and it should show something like this:

```
(floris) >>~ $ pip freeze | grep numexpr
numexpr==2.8.1
```
